File: src/main/java/bad/robot/radiate/teamcity/FailedToCreateYmlFile.java
package bad.robot.radiate.teamcity;

import bad.robot.radiate.monitor.Information;

import static java.lang.String.format;
import static org.apache.commons.lang3.StringUtils.defaultString;

class FailedToCreateYmlFile extends Information {

	public FailedToCreateYmlFile(Exception e) {
		super(format("Failed to create Yml configuration file (caused by an %s %s), falling back to use environment variables", e.getClass().getSimpleName(), defaultString(e.getMessage())));
	}
}

File: src/main/scala/bad/robot/radiate/teamcity/FailedToCreateYmlFile.scala
package bad.robot.radiate.teamcity

import bad.robot.radiate.monitor.Information
import bad.robot.radiate.teamcity.StringOps._

class FailedToCreateYmlFile(e: Exception) extends Information(
  s"Failed to create Yml configuration file (caused by an ${e.getClass.getSimpleName} ${defaultString(e.getMessage)}), falling back to use environment variables"
)

object StringOps {
  def defaultString(string: String): String = {
    if (string == null) "" else string.toString
  }
}
